### DISCUSSION TITLE
Project Update December 2022

### DISCUSSION THREAD

**Discussion by DeeEmm** [DIY-Flow-Bench/DIY-Flow-Bench#109] [Announcements]
A new month and a new update.

Build https://github.com/DeeEmm/DIY-Flow-Bench/commit/9c829c81392a61dfded554dbd3d9fb997ff70fe6 is the result of last months code push to get the code to a point where it was stable and functioning correctly. The I2C and MAF issues have now been addressed and apart from the occasional web / SSE connection drop things are pretty good.

The MAF Data files have been updated so if you are using a MAF sensor that we have the file for it should hopefully work. There is one caveat to this in that support for frequency based sensors is not currently functional and needs work to get up and running.

Next stage for me is to get the shield tested with the new code and get that ready for a small production run.

I will tidy up the code and get it packaged up as an official release in the next few days. However if you cannot wait that long then the commit linked to above should work fine but you will need to change the board type in the config settings.

As always please give feedback / bug reports - either here or raise a bug in the issues list.

/DM

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/a9510c549f3ba0a4e012699d5fd56e16783d7f22 has minor code tidying, Doxygen style headers, removal of unused vars etc.

I have added a pin definition for the Arducam board but still need to set the pin allocation as this is just a copy and paste of the Wemos pindef. We now have pin definitions for the Wemos D1R32 / Arducam ESP32S / Generic Espiduino / ESP32_WROVER_KIT Development board

I've moved from testing on the ESP32_WROVER_KIT Development board to the Wemos D1R32 and have noted that there are some issues not present on the dev board. Namely that initial wifi connection is a bit flakey and takes several attempts to connect to my Wifi rather than timing out and creating an Access Point. I suspect that this is related to the task handling and can hopefully be rectified by tweaking task priorities.

Similarly I am also seeing the message 'ERROR: Too many messages queued ' in the serial monitor. This message is generated by the Webserver library and I suspect is also related to the task handling priority / timing.

There's a bunch of settings I can play with there so hopefully it should not be too hard to resolve.

I'll keep chipping away at this until I've got these issues resolved.

/DM

**Reply by DeeEmm** [reply]
Created bug #110 [BUG] ERROR: Too many messages queued

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/6b1a2e4eeae27d33a8ceb6cb0f4817cdef32e66e fixes ADC disable and has some other code tidying.

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/7fd547915a78ec22b1302372bc8705cbd1587b81 addresses bug #111.

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/82b7c9a11819b4ee70b7615198ff557c415bdf35 improves Wifi Connection but I am still seeing some issues with connectivity. There are some known connectivity issues with ESPAsyncWebserver which I have tried to address with known published workarounds. 

This issue has unfortunately been burning up a little time and taking me away from finishing the rest of the unit testing.

I'm interested to hear if anyone else is experiencing WiFi issues as at least some of the known issues relate to environmental (router) based constraints. 

The two issues I'm seeing are failure to instantiate initial WiFi connection and then once connected browser connection persistence.

I think I'll leave the Wifi where it is at the moment although I suspect that I will need to revisit it at a later date.

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/c831c2b2295fcc04d5f2d3081294691d333821b5 Hopefully addresses the connectivity issues. I have moved the SSE task back into the main loop and combined the separate SSE events into one which has massively stabilised connectivity. There's some other minor updates and code cleaning.

Definitely going to move on from WiFi now. 🤪 

Unfortunately however it is now the silly season, xmas is nearing which means that work is winding up for me and I'm winding down into family stuff so I might not get much of a chance over xmas to get the units tests finished off. If I get any more work done I'll post here, but other than that the next major update will most likely be the new year.

**Comment by DeeEmm** [comment]
Status Update:

Commit https://github.com/DeeEmm/DIY-Flow-Bench/commit/31bd75b5060e5e4ee337d3f3975db7e8dafa8cfe adds a couple of features - ability to change temperature units and also adjusted flow which closes out feature request #8 . This allows user to view flow in alternate depressions. For example if your bench can only achieve 15"wg depression but you want to view flow figures at 28" depression the adjusted flow value shows this in real time. 

The adjusted flow value can be viewed by clicking on the flow value in the GUI. This toggles between flow and adjusted flow.

There's also some other minor code tidying

**Comment by dobrowolskibartosz** [comment]
So cool to see more updates, in meantime I'll try to find a reliable, popular in europe MAF and all the required specs ( this will be difficult) as LS's are a luxury here in the old land. Some bimmer or vag should to the trick

**Reply by DeeEmm** [reply]
The LS MAFs are pretty generic and used on a variety of cars but I'm not sure that any made it across the pond. 

Ultimately it will be possible to map MAF sensors using a known MAF but that requires a calibrated and working bench. When I get to a point where everything is stable I will be looking into this as many people are interested repurposing scrapyard components. If I can provide some maps for generic common vehicles then anyone should be able to source a cheap MAF sensor from their local wrecking yard.
